component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateUtil.java

```diff
@@ -382,20 +382,27 @@ private static boolean isUserExists(String userName, AuthenticationContext authe
     /**
      * Check whether user account is locked or not.
      *
-     * @param user Authenticated user.
+     * @param subject userName that passed from the cert CN value.
      * @return boolean account locked or not.
      * @throws AccountLockServiceException
      */
-    public static boolean isAccountLock(AuthenticatedUser user) throws AccountLockServiceException {
+    public static boolean isAccountLock(String subject) throws AccountLockServiceException {
+        //Get the tenantaware username & particular tenant domain
+        String userName = null;
```

Comment:
if the subject doesn't tenant domain then userName will be null according to the logic? But ideally, userName should be equal to the subject if the subject doesn't have tenant domain. So please change it to following.

```suggestion
        String userName = subject;
```